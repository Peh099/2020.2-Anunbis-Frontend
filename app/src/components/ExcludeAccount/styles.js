import styled from 'styled-components';

export const ContentExclude = styled.div`
  position: absolute;
  width: 250px;
  height: 100px;
  display: flex;
  flex-direction: column;
  background-color: var(--lightWhite);
  margin-top: 150px;
  border-radius: 10px;
  
  header {
    border-top-right-radius: 10px;
    border-top-left-radius: 10px;
    display: flex;
    background-color: #212121;
    color: white;
    text-align: center;
    font-size: 15px;
    height: 30px;
    align-items: center;
    justify-content: center;
  }
`;

export const BtsExclude = styled.div`
  display: flex;
  flex-direction: row;
  background-color: inherit;
  height: inherit;
  justify-content: center;
  align-items: center;
  border-radius: 10px;

  Button {
    color: white;
    font-weight: bold;
    border: none;
  }
`;