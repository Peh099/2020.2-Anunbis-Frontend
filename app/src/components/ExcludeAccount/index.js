import React from 'react';
import { useHistory } from 'react-router';
import Users from '../../services/Users';
import Button from '../Button';
import { ContentExclude, BtsExclude } from './styles';
import api from '../../services/Api';
import { logOut } from '../../services/Auth';

export default function ExcludeAccount(props) {
  const history = useHistory();

  function deleteAcc() {
    api.delete(`${Users.whoAuthenticated().localStorageName}`).then(() => {
      logOut();
      history.push('/');
    });
  }

  return (
    <ContentExclude>
      <header>DESEJA EXCLUIR SUA CONTA?</header>
      <BtsExclude>
        <Button
          backColor="#26A69A"
          padding="10px 10px"
          text="VOLTAR"
          onClick={() => props.close()}
        />
        <Button
          backColor="var(--lightRed)"
          padding="10px 7px"
          text="EXCLUIR"
          onClick={() => deleteAcc()}
        />
      </BtsExclude>
    </ContentExclude>
  );
}
