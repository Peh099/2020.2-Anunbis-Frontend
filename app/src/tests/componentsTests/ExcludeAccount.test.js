import renderer from 'react-test-renderer';
import ExcludeAccount from '../../components/ExcludeAccount';

describe('Snapshot ExcludeAccount component', () => {
    it('matches the snapshot', () => {
        const tree = renderer.create(<ExcludeAccount />).toJSON()
    
    expect(tree).toMatchSnapshot()
    })
})