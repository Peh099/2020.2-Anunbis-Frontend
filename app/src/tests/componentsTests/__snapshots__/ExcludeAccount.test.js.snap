// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot ExcludeAccount component matches the snapshot 1`] = `
<div
  className="sc-gtsrHT dZaWmq"
>
  <header>
    DESEJA EXCLUIR SUA CONTA?
  </header>
  <div
    className="sc-dlnjwi fUxTvQ"
  >
    <button
      className="sc-bdnxRM jfukfz"
      onClick={[Function]}
    >
      VOLTAR
    </button>
    <button
      className="sc-bdnxRM fDUycc"
      onClick={[Function]}
    >
      EXCLUIR
    </button>
  </div>
</div>
`;
